### ISSUE TITLE
List tables 4 life

### ISSUE THREAD

**Issue by moigagoo** [mkdocs/mkdocs#62]
TL;DR: List tables are cool and extremely useful, if only MkDocs could support them.
## Problem

Currently, there's one way to create tables in Markdown—visually mimicking the table with bars and other characters:

```
| First Header  | Second Header |
| ------------- | ------------- |
| Content Cell  | Content Cell  |
| Content Cell  | Content Cell  |
```

This approach works OK only with primitive tables with little content.

For example, consider a table with columns Title, Description, and Default value. Title is probably a single word, Description is a paragraph, and Default value is a number.

In Markdown, it could look something like this:

```
| Title | Description | Default value |
| ---- | ----------- | ------------- |
| Foo | Lorem ipsum dolor amet, blah-blah-blah, really long description, just goes on and on | 42 |
| Bar | Short description | 2 |
| Baz | A very long description again. Notice how the table has already completely fallen apart because you cannot track the table's structure anymore since all lines have different length. | 13 |
```

This table has problems: 1) it doesn't look as a table anymore, 2) it's barely maintainable, and 3) it's ugly :-(
## Solution

List tables! Here's an example from ReST:

.. list-table::
    :header-rows: 1

```
*   -   Title
    -   Description
    -   Default value
*   -   Foo
    -   Lorem ipsum dolor amet, blah-blah-blah, really long description, just goes on and on
    -   42
*   -   Bar
    -   Short description 
    -   2
*   -   Baz
    -   A very long description again. Notice how the table has already completely fallen apart because you cannot track the table's structure anymore since all lines have different length.
    -   13
```

Basically, it's a list of lists, which is much more readable and sparse. It can be easily extended.
## Conclusion

There should be a Markdown extension for list tables.

MkDocs repo is probably not the best place to report this sort of feature requests, still, MkDocs could definitely benefit from having such a feature, since bigger projects could migrate from Sphinx.

Thanks!
